lib/rex/ui/text/shell/history_manager.rb

```diff
@@ -91,47 +99,78 @@ def readline_available?
     defined?(::Readline)
   end
 
+  def reline_available?
+    begin
+      require 'reline'
+      defined?(::Reline)
+    rescue ::LoadError => _e
+      false
+    end
+  end
+
   def clear_readline
     return unless readline_available?
 
     ::Readline::HISTORY.length.times { ::Readline::HISTORY.pop }
   end
 
-  def load_history_file(history_file)
-    return unless readline_available?
+  def clear_reline
+    return unless reline_available?
 
-    clear_readline
```

Comment:
May need `clear_readline` after all?

Reply:
I changed this line to `context[:input_library] == :reline ? clear_reline : clear_readline` and moved it into the block after we have successfully opened the new file:
```ruby
File.open(history_file, 'r') do |f|
        context[:input_library] == :reline ? clear_reline : clear_readline
        ...
```
Fixed 👍